src/main/java/org/intellij/sdk/codesync/clients/JSONResponse.java

```diff
@@ -78,10 +78,13 @@ public void raiseForStatus () throws StatusCodeError {
                 if (errorObject != null && errorObject.containsKey("message")) {
                     errorMessage = (String) errorObject.get("message");
                 }
-                if (errorObject != null && errorObject.containsKey("error_code")) {
-                    errorCode = ((Number) errorObject.get("error_code"));
-                    if (errorCode != null){
-                        customErrorCode = errorCode.intValue();
+                // Extract error_code only if a 402 is received
+                if(this.statusCode == 402){
+                    if (errorObject != null && errorObject.containsKey("error_code")) {
+                        errorCode = ((Number) errorObject.get("error_code"));
+                        if (errorCode != null) {
```

Comment:
Do we need this logic now?

Reply:
Technically we don't since we are always sending an error_code for a 402, but i was thinking we keep it to be on the safe side. What do you think?

Reply:
Sure, sounds good